### PR TITLE
5 uh 5 walletwallet

### LINKED ISSUE #5: [UH: 5] /wallet/{wallet_id}
Desc: Gets wallet cryptocurrencies
Expected Result:
[{
      "coid_id": "string",
      "name": "string",
      "symbol": "string",
      "amount": 0,
     "value_usd": 0 
   }]

### PR DESCRIPTION
**Pull request by ivans1516**

### CODE REVIEW

**Comment by jalex3421** [comment]
Unit testing should be also done

File: app/Application/CoinLoreCryptoDataSource/CoinLoreCryptoDataSource.php [hunk 1]
```diff
@@ -3,7 +3,7 @@
 namespace App\Application\CoinLoreCryptoDataSource;
 use App\Domain\Coin;
 
-Interface CoinLoreCryptoDataSource
+Interface CoinLoreCryptoDataSource//cambiar nombre
```
Comment (jalex3421) [inline app/Application/CoinLoreCryptoDataSource/CoinLoreCryptoDataSource.php]: The name of the class should be replaced for a a more general one.

File: app/Infrastructure/Cache/WalletCache.php [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 class WalletCache
 {
-    public function open():string
+    public function open():string//servicio
```
Comment (jalex3421) [inline app/Infrastructure/Cache/WalletCache.php]: This method might be better if we change the open method location.

File: app/Infrastructure/Cache/WalletCache.php [hunk 3]
```diff
@@ -36,7 +36,7 @@ public function set(string $wallet_id, Wallet $wallet): void
         Cache::put('wallet'.$wallet_id,$wallet,600);
     }
```
Comment (jalex3421) [inline app/Infrastructure/Cache/WalletCache.php:30]: Maybe the name of the variable is better if we make the following change:
$wallet_id ---> $walletId

**Comment by jalex3421** [comment]
By the way, other changes should be done before merging this pull request. The desired changes are the following:
-  Some tests that involve the use of  the Cache should be use a mockery approach.
- The name of the variables should follow the pattern: $thisIsAVariable.
- If it is possible , a lavarel cache test could be added to the whole project. 
- Parameters validation --> Parameters validator
- Test the cache.
- getCoin method  should separate responsibilities.

Come on team!! @ivans1516 @crackergirl

**Comment by crackergirl** [comment]
In many classes, it is necessary to repeat the tests because I have modified the cache and created an interface. Also, there were services that called the coin lore and the cache and a service has to call one of them, it is the controller that calls the services.

**Pull request merged**